ERCS/erc-chain_specific_addresses.md

```diff
@@ -0,0 +1,235 @@
+
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Format
+
+Valid addresses MUST include the identifier of the chain they belong to.
+
+```
+L1-TLD ::= eth | sepolia | …
+chain_id ::= 0x[a-fA-F0-9]{1,64}
+address ::= 0x[a-fA-F0-9]{40}
+chain ::= <chain_id> | <L1-TLD> | <ens-name> . <L1-TLD>
+user ::= <address> | <ens-subdomain>
+account ::= <user>@<chain>
+```
+
+Note the difference between `ens-name`, which is a full ENS name, and `ens-subdomain` that is just a segment of a name between dots. E.g. `user.app.eth` is a name, `user` and `app` are subdomains.
+
+A few examples below. 
+
+Option 1: using @ to separate address and chain
+```
+Mainnet
+- 0x12345...6789@0x1
+- 0x12345...6789@eth
+- alice.eth@eth
+
+Testnet (Sepolia)
+- 0x12345...6789@0xaa36a7
+- 0x12345...6789@sepolia
+- alice.eth@sepolia
+
+Rollup
+- 0x12345...6789@chainId
+- 0x12345...6789@arbitrum.eth
+- alice.eth@arbitrum.eth
+
+Rollup Sepolia
+- 0x12345...6789@arbitrum.sepolia
+
+My ENS name is registered on rollup1, but I want to receive funds on rollup2
+- alice.rollup1.eth@rollup2.eth
+```
+
+Option 2: using : instead of @ 
+```
+Mainnet
+- 0x12345...6789:0x1
+- 0x12345...6789:eth
+- alice.eth:eth
+
+Testnet (Sepolia)
+- 0x12345...6789:0xaa36a7
+- 0x12345...6789:sepolia
+- alice.eth:sepolia
+
+Rollup
+- 0x12345...6789:chainId
+- 0x12345...6789:arbitrum.eth
+- alice.eth:arbitrum.eth
+
+Rollup Sepolia
+- 0x12345...6789:arbitrum.sepolia
+
+My ENS name is registered on rollup1, but I want to receive funds on rollup2
+- alice.rollup1.eth:rollup2.eth
+```
+### CHECKSUM
+TODO: add more explanation here
+
+Two desired properties:
+1) checksums are MANDATORY
+2) checksum hash goes over the entire address, so users can't just go and replace a component and expect it to stay valid
+
+
+### A special case for ENS resolution
+
+Any ENS name today can be resolved to a chain identifier as in [ENSIP-11](https://docs.ens.domains/ensip/11) or to an address as in [ENSIP-9](https://docs.ens.domains/ensip/9).
+We could imagine having a name `user.eth` that points to a record of the form `{address ; chain_id}`. Given such an address a wallet can verify it resolves to a valid account.
+The advantage of this format is that it is very flexible and can accommodate a number of use cases, however it can also lead to confusion for users because a name does not necessarily resolve to a valid account. The same `user.eth` could lead to a website, a NFT or multiple addresses.
+
+The resolution of a `address@chain` on the contrary, imposes that the left-hand resolves to an address and the right-hand to a chain identifier.
+
+When given a `user@rollup.eth`, the wallet can resolve `rollup.eth` to get a chain identifier and `user.rollup.eth` to get an address. In any other case it fails.
```

Comment:
This doesn't require a new syntax: When the wallet is resolving `user.rollup.eth`, the standard the resolution process should be resolve to resolve the way you described: resolve `rollup.eth` to get a chain identifier and `user.rollup.eth` to get an address.

This approach also allows `user.rollup.eth` to override this behavior by setting a record for their own preferred chain identifier to use when their ENS name is resolved. `*.rollup.eth` names will likely never override this, but I'd expect almost every `user.eth` to want to receive funds on some chain that isn't L1.